src/charm.py

```diff
@@ -387,23 +387,23 @@ def _set_up_discourse(self, event: HookEvent) -> None:
                     [f"{DISCOURSE_PATH}/bin/bundle", "exec", "rake", "--trace", "db:migrate"],
                     environment=env_settings,
                     working_dir=DISCOURSE_PATH,
-                    user="discourse",
                 )
                 migration_process.wait_output()
             self.model.unit.status = MaintenanceStatus("Compiling assets")
-            precompile_process: ExecProcess = container.exec(
-                [f"{DISCOURSE_PATH}/bin/bundle", "exec", "rake", "assets:precompile"],
-                environment=env_settings,
-                working_dir=DISCOURSE_PATH,
-                user="discourse",
-            )
-            precompile_process.wait_output()
+            try:
+                precompile_process: ExecProcess = container.exec(
```

Comment:
Perhaps for a different PR, I remember someone mentioned that precompile isn't strictly necessary, if the assets aren't precompiled that will happen at runtime. Something to think about is to do this after going into active so that the startup time is sped up

Reply:
Do you have a ref for this so that I can create the corresponding issue ?

Reply:
No reference, perhaps this would be a spike to investigate whether it is possible to skip the precompile action and the tests still pass? And see what the impact is to the user experience

Reply:
I created a spike for this: https://warthogs.atlassian.net/browse/ISD-1001